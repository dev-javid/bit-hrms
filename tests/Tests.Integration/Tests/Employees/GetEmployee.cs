namespace Tests.Integration.Tests.Employees
{
    public class GetEmployee(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/employees/3";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            // Arrange
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Employees/GetEmployee.sql");

            // Act
            using HttpResponseMessage response = await Client.GetAsync(Route);

            // Assert
            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
