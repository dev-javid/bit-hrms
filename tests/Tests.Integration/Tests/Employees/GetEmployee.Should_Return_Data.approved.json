{
    "employeeId": 3,
    "departmentId": 3000,
    "departmentName": "Test-Department",
    "jobTitleId": 4000,
    "jobTitleName": "Test-Department",
    "firstName": "Test-First-3",
    "lastName": "Test-Last-3",
    "fullName": "Test-First-3 Test-Last-3",
    "dateOfBirth": "2000-05-03",
    "dateOfJoining": "2024-05-03",
    "fatherName": "Test-Father-3",
    "phoneNumber": "01234567899",
    "companyEmail": "test3@example.com",
    "personalEmail": "test6@example.com",
    "address": "Test-Address",
    "city": "Test-City",
    "pan": "XXXXX0000X",
    "aadhar": "000000000000",
    "documents": []
}